p2p/netutil/net.go

```diff
@@ -299,10 +305,15 @@ func (s *DistinctNetSet) key(ip net.IP) net.IP {
 func (s DistinctNetSet) String() string {
 	var buf bytes.Buffer
 	buf.WriteString("{")
-	keys := make([]string, 0, len(s.members))
-	for k := range s.members {
-		keys = append(keys, k)
+	if s.members == nil {
+		s.members = &sync.Map{}
+		s.buf = make(net.IP, 17)
```

Comment:
```suggestion
        return "{}"
```
if the map doesn't exist - return an empty representation